### DISCUSSION TITLE
Problem running NextSIM-DG on GRICAD

### DISCUSSION THREAD

**Discussion by stephanieleroux** [nextsimdg/nextsimdg#680] [General]
Francesca Vittorioso @vittofra contacted Aurélie and Stephanie on Aug 27th 2024:
>27/08/2024 à 13:23, Francesca Vittorioso a écrit :
> Good morning to both of you,
> After several months, I am back to running nextsimdg on Gricad. However, I am encountering an issue with the compilation. The compiler is not present. Could you help me with this?

**Comment by stephanieleroux** [comment]
We  have investigated the problem. @vittofra: you'll need to re-install your environment and re-compile the code following the tuto that Aurélie is updating. More soon!
@auraoupa and @stephanieleroux

**Reply by auraoupa** [reply]
So in order to make nextsimdg run on gricad you need the right combination of nextsimdg version and conda environment, one that is currently working fine is : 
  - the nextsim_dg version updated on 2024, January, right after @timspainNERSC 's visit to Grenoble, when the tutorial for running nextsimdg on gricad has been designed, it can be obtained with : ```git clone -b develop https://github.com/nextsimhub/nextsimdg.git nextsimdg; cd nextsimdg; git checkout 493ec8fb```
  - the conda environment built with [this specified list of librairies](https://ige-meom-opendap.univ-grenoble-alpes.fr/thredds/catalog/meomopendap/extract/SASIP/model-configurations/neXtSIM-DG/demo-june2023/catalog.html?dataset=meomscanpublic/SASIP/model-configurations/neXtSIM-DG/demo-june2023/spec-file-SLX.txt), you can build it on gricad with : ```conda create --name nextsimdg --file /summer/sasip/model-configurations/neXtSIM-DG/demo-june2023/spec-file-SLX.txt````

The [tutorial on running the June 2023 demo](https://github.com/sasip-climate/GRICAD-usage/blob/main/netxsimdg_GRICAD.md) has been updated to reflect this.

Hope it will work for you too @vittofra ?
Next step, I will check with a more recent version of the develop branch !

**Comment by auraoupa** [comment]
Update : the actual develop branch (commit 6eedb7e8) can also run with this environment, no need to modify config_june23.cfg either !

**Reply by vittofra** [reply]
Hello @auraoupa and @stephanieleroux and thank you for the updates. 
I managed to install and compile the model following your indications. 

However, I wanted to point out that, compared to what was specified in the tutorial, in order to launch a job in the devel option, it is necessary to connect to dahu3.  Please correct me if I am wrong.

**Reply by aureliealbertmeom** [reply]
Yes you are right, I will correct it indeed !

**Comment by yumengch** [comment]
Hi Aurelie, Hi Stephanie,

Thank you for update the environment for nextsimdg. I post the issue here as I think it is relevant. 

I followed steps given in the instructions, the model compiles, but I get the following error message when I try to run the model:
```terminate called after throwing an instance of 'netCDF::exceptions::NcHdfErr'
  what():  NetCDF: HDF error
file: /home/conda/feedstock_root/build_artifacts/netcdf-cxx4_1704389085991/work/cxx4/ncVar.cpp  line:1049
./test.sh: line 15: 10100 Aborted                 ./nextsim --config-file config.cfg > time.step
```
This occurs when I run it both through `oarsub`, or directly from commandline. It will be helpful if you could provide a hint on the possible reasons.

Cheers,
Yumeng

**Reply by stephanieleroux** [reply]
Hi Yumeng,
It worked for 3 of us a week ago with the develop branch (commit ([`6eedb7e`](https://github.com/nextsimhub/nextsimdg/commit/6eedb7e827ada2a3f5a51864c4043f6946e7d822)) and this exact list of [packages](https://ige-meom-opendap.univ-grenoble-alpes.fr/thredds/catalog/meomopendap/extract/SASIP/model-configurations/neXtSIM-DG/demo-june2023/catalog.html?dataset=meomscanpublic/SASIP/model-configurations/neXtSIM-DG/demo-june2023/spec-file-SLX.txt). In particular it must include netcdf4.
Are you working in a conda environment installed with this exact list of packages? (following Aurélie's above instructions)
The develop branch might have changed since our last test. I'll try to reproduce your problem later this week.
More soon.
Best,

**Reply by yumengch** [reply]
Hi Stephanie,

I finally understand the problem... It is not about the environment or the model code. It seems that it's because I used the wrong config file.

In `/summer/sasip/model-configurations/neXtSIM-DG/demo-june2023/config_june23.cfg`, taking the `DiagnosticOutputModule` as an example. In summer storage, `Nextsim::IDiagnosticOutput =  Nextsim::ConfigOutput` is used in the `Modules` section while in the repository examples, `DiagnosticOutputModule =  Nextsim::ConfigOutput` is used. 

I guess the reason that the HDF5 error is raised was because the model was not properly ran and the potentially uninitialised/allocated values were trying to be written into the file. 

Cheers,
Yumeng
